src/sentry/ratelimits/redis.py

```diff
@@ -106,8 +106,11 @@ def is_limited_with_value(
         # Reset Time = next time bucket's start time
         reset_time = _bucket_start_time(_time_bucket(request_time, window) + 1, window)
         try:
-            result = self.client.incr(redis_key)
-            self.client.expire(redis_key, expiration)
+            pipe = self.client.pipeline()
+            pipe.multi()
```

Comment:
It seems like the pipeline is the more important part here, rather than the multi.

I'm not sure how well `.multi` works with `redis-py-cluster`. And I think in CI we're not actually using redis cluster at all, so if it doesn't work, it might just break in production.

Reply:
Looking at the code, I think this will actually break https://github.com/Grokzen/redis-py-cluster/blob/8a8102a9d758d61a7ec1e2ac9050fcd34029ff3f/rediscluster/pipeline.py#L281

Reply:
It seems redis-py-cluster is archived. Maybe we should move to a more maintained solution.

Reply:
Yeah... Seems bad that we're using something unmaintained. I think swapping it out is probably pretty scary too

Reply:
You chose the correct words to motivate me. Thank you! I'll look into it.

Reply:
@wedamija we have some preparation for running redis-cluster in dev/ci but no test is using it afaik
https://github.com/getsentry/sentry/pull/49606/files#diff-94dc6dcb2c82ecd1b4fab0edee177a697e821f1340bd6710392b4095880a0e87R60

Reply:
I'm working on removing `redis-py-cluster` at: https://github.com/getsentry/sentry-redis-tools/pull/9

Reply:
Two weeks have passed, and finally we've updated Redis to v4. https://github.com/getsentry/sentry/pull/65282